src/interfaces/chain.h

```diff
@@ -135,6 +105,28 @@ class Chain
     //! Get block hash. Height must be valid or this function will abort.
     virtual uint256 getBlockHash(int height) = 0;
 
+    //! Check that the block is available on disk (i.e. has not been
+    //! pruned), and contains transactions.
+    virtual bool haveBlockOnDisk(int height) = 0;
```

Comment:
In commit "[wallet] Move methods from Chain::Lock interface to simple Chain" (22d0fb127e99d6cdcaa003785a7278f58038b465)

Note: After this commit, the only remaining Chain::haveBlockOnDisk, findFirstBlockWithTimeAndHeight, getTipLocator, and findLocatorFork calls are in CreateWalletFromFile and are removed in #15719. I expect this PR will get merged before #15719, so just making a note about this.

Similarly, the Chain::checkFinalTx method can go away after #17443, which removes all those calls